packages/cli/src/credentials/credentials.service.ts

```diff
@@ -36,4 +96,162 @@ export class CredentialsService {
 		}
 		return new Credentials({ id: id.toString(), name }, type, nodesAccess, data);
 	}
+
+	static async prepareCreateData(
+		data: CredentialRequest.CredentialProperties,
+	): Promise<CredentialsEntity> {
+		// Make a copy so we can delete the provided ID and
+		// so we can modify nodesAccess
+		const dataCopy = clone(data);
+		delete dataCopy.id;
+
+		const newCredentials = new CredentialsEntity();
+		Object.assign(newCredentials, dataCopy);
```

Comment:
this saves us the merge
```suggestion
		const newCredentials = Db.collections.Credentials.create(dataCopy);
```